apps/balancer-tools/src/app/poolsimulator/(components)/Menu.tsx

```diff
@@ -26,8 +31,82 @@ const POOL_TYPES_MAPPER = {
 };
 
 function IndexMenu() {
+  const { push } = useRouter();
+  const { setCustomData, customData, setIsGraphLoading } = usePoolSimulator();
+  const [tabValue, setTabValue] = useState("initialData");
+  const clickInitialDataTab = new CustomEvent("clickInitialDataTab");
+  const clickCustomDataTab = new CustomEvent("clickCustomDataTab");
+
+  const initialFormExtraOnSubmit = (data: AnalysisData, _: boolean) => {
+    if (!customData.poolParams) {
+      setCustomData(data);
+    }
+    setTabValue("customData");
+  };
+
+  const customFormExtraOnSubmit = (_: AnalysisData, tabClicked: boolean) => {
+    if (!tabClicked) {
+      setIsGraphLoading(true);
+      push("/poolsimulator/analysis");
+    }
+    setTabValue("initialData");
+  };
+
+  return (
+    <div>
+      <Tabs value={tabValue} defaultValue="initialData">
+        <Tabs.ItemTriggerWrapper>
+          <Tabs.ItemTrigger
+            tabName="initialData"
+            color="blue7"
+            onClick={() => {
+              document.dispatchEvent(clickInitialDataTab);
+            }}
+          >
+            <span>Initial</span>
+          </Tabs.ItemTrigger>
+          <Tabs.ItemTrigger
+            tabName="customData"
+            color="amber9"
+            onClick={() => {
+              document.dispatchEvent(clickCustomDataTab);
+            }}
+          >
+            <span>Custom</span>
+          </Tabs.ItemTrigger>
+        </Tabs.ItemTriggerWrapper>
+        <Tabs.ItemContent tabName="initialData">
+          <InitialFormContextProvider>
+            <SearchPoolFormWithDataForm>
+              <div className="flex flex-col mt-4">
+                <FormWithPoolType extraOnSubmit={initialFormExtraOnSubmit} />
```

Comment:
```suggestion
                <FormWithPoolType onSubmit={initialFormExtraOnSubmit} />
```

Reply:
I just want to make clear that we already have another `onSubmit` on the `FormWithPoolType` component and this function will just extend the existing function.